--- a/build.info
+++ b/build.info
@@ -138,0 +139 @@
+    libdir= CMAKECONFIGDIR= PKGCONFIGDIR= \
--- a/exporters/build.info
+++ b/exporters/build.info
@@ -21,0 +22 @@
+    COMMENT="This file provides configuration information for OpenSSL" \
--- a/util/mkinstallvars.pl
+++ b/util/mkinstallvars.pl
@@ -21,0 +22,2 @@
+my $debug = $ENV{OPENSSL_MKINSTALLVARS_DEBUG} || 0;
+
@@ -38 +40 @@
-print STDERR "DEBUG: all keys: ", join(", ", sort keys %all), "\n";
+print STDERR "DEBUG: all keys: ", join(", ", sort keys %all), "\n" if $debug;
@@ -55 +57 @@
-print STDERR "DEBUG: LIBDIR = $values{LIBDIR}->[0], libdir = $values{libdir}->[0] => ";
+print STDERR "DEBUG: LIBDIR = $values{LIBDIR}->[0], libdir = $values{libdir}->[0] => " if $debug;
@@ -57 +59 @@
-print STDERR "LIBDIR = $values{LIBDIR}->[0]\n";
+print STDERR "LIBDIR = $values{LIBDIR}->[0]\n" if $debug;
@@ -72 +74 @@
-    print STDERR "DEBUG: $k = $v->[0] => ";
+    print STDERR "DEBUG: $k = $v->[0] => " if $debug;
@@ -75 +77 @@
-    print STDERR "$k = $v->[0]\n";
+    print STDERR "$k = $v->[0]\n" if $debug;
@@ -93 +95 @@
-            " => ";
+            " => " if $debug;
@@ -112 +114 @@
-            "\n";
+            "\n" if $debug;
